numpy/core/src/multiarray/nditer_api.c

```diff
@@ -1390,6 +1390,47 @@ NpyIter_GetInnerLoopSizePtr(NpyIter *iter)
     }
 }
 
+/*NUMPY_API
+ * Resolves all writebackifcopy scratch buffers, not safe to use iterator
+ * operands after this call, in this iterator as well as any copies.
+ * Returns 0 on success, -1 on failure
+ */
+NPY_NO_EXPORT int
+NpyIter_Close(NpyIter *iter)
```

Comment:
Is it safe to call this multiple times?

Reply:
yes

Reply:
the first call will resolve all writebacks, so if there was a writeback scratch buffer now the iterator operand's `base` attribute will be `None` and the operand's `data` will be independent from the original writeable array `data`. Subsequent calls will no do anything